cura/Machines/Models/QualitySettingsModel.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+from PyQt5.QtCore import pyqtProperty, pyqtSignal, Qt
+
+from UM.Application import Application
+from UM.Qt.ListModel import ListModel
+from UM.Settings.ContainerRegistry import ContainerRegistry
+
+
+class QualitySettingsModel(ListModel):
```

Comment:
Doc added in b5cf6b7531596a38171d6a561693040eb484fac3. It is hard to use the PropertyProviders because they cannot retrieve the correct layer height if a qualtiy/custom quality that's not activated at the moment is selected.